libass/ass.c

```diff
@@ -629,6 +629,7 @@ static int process_styles_line(ASS_Track *track, char *str)
         skip_spaces(&p);
         free(track->style_format);
         track->style_format = strdup(p);
+        if (!track->style_format) return -1;
```

Comment:
Inconsistent error reporting: some `return -1` branches have dedicated messages, some don't.
Each function should be in one of the two classes:
- Do not report errors, only return failure flags. It's caller responsibility to report errors.
- Report all errors itself. Caller should not do any additional reporting.

Reply:
With the exception of `STRVAL` and `STARREDSTRVAL` for above mentioned reasons, all parsing related errors should now be reported by the caller, while errors during rendering are directly reported.

Reply:
Same.